pkg/cloudprovider/providers/gce/gce_disks.go

```diff
@@ -555,14 +532,7 @@ func (gce *GCECloud) AttachDisk(diskName string, nodeName types.NodeName, readOn
 		readWrite = "READ_ONLY"
 	}
 
-	attachOp, err := gce.manager.AttachDiskOnCloudProvider(
-		disk, readWrite, instance.Zone, instance.Name)
-
-	if err != nil {
-		return mc.Observe(err)
-	}
-
-	return gce.manager.WaitForZoneOp(attachOp, instance.Zone, mc)
```

Comment:
We used to WaitForZoneOp with retry interval and timeout, is that captured in the generated API? Ditto everywhere else

Reply:
Each client call blocks until it's accepted by a RateLimiter. A general RateLimiter is used across GCE cloud provider and is initialized here:

https://github.com/kubernetes/kubernetes/blob/e5274b6376b8da64b4e784c742585270cc1d27ca/pkg/cloudprovider/providers/gce/gce.go#L487

So the query is polled every 100ms instead of every 3sec. I'm not sure if there is a timeout, though.

cc @bowei

Reply:
Looking at it again, there doesn't seem to be retries for the generated API calls

Reply:
The ratelimiter does handle the interval... I'll be making some changes so it's not so frequent, but it should be invisible to your changes.

As discussed with verult@, timeout is implemented via context with a timeout.